File: chtl_compiler/merger/result_merger.py
from __future__ import annotations

from chtl_compiler.types import CompilationBundle


class ResultMerger:
	def __init__(self) -> None:
		pass

	def merge_to_html_document(self, bundle: CompilationBundle) -> str:
		unit = bundle.merge()
		css_block = f"<style>\n{unit.css}\n</style>\n" if unit.css.strip() else ""
		js_block = f"<script>\n{unit.js}\n</script>\n" if unit.js.strip() else ""
		body = unit.html
		# If result already includes <html>, assume full doc
		if '<html' in body or '<!DOCTYPE' in body:
			# Append assets if not already present (naive)
			return f"{body}\n{css_block}{js_block}".strip()
		# Minimal HTML doc scaffold
		return (
			"<!DOCTYPE html>\n"
			"<html lang=\"en\">\n"
			"<head>\n<meta charset=\"utf-8\">\n" + css_block + "</head>\n"
			"<body>\n" + body + "\n" + js_block + "</body>\n"
			"</html>\n"
		)

File: src/main/java/com/example/chtl/merge/ResultMerger.java
package com.example.chtl.merge;

import com.example.chtl.dispatch.CompilationResult;

public class ResultMerger {
	public static String mergeToHtmlDocument(CompilationResult result) {
		StringBuilder sb = new StringBuilder();
		sb.append("<!DOCTYPE html>\n");
		sb.append("<html lang=\"zh-CN\">\n");
		sb.append("<head>\n");
		sb.append("  <meta charset=\"UTF-8\"/>\n");
		sb.append("  <meta name=\"viewport\" content=\"width=device-width, initial-scale=1.0\"/>\n");
		if (!result.getGlobalCss().isBlank()) {
			sb.append("  <style>\n");
			sb.append(result.getGlobalCss());
			sb.append("  </style>\n");
		}
		sb.append("</head>\n");
		sb.append("<body>\n");
		sb.append(result.getHtmlBody()).append('\n');
		if (!result.getGlobalJs().isBlank()) {
			sb.append("  <script>\n");
			sb.append(result.getGlobalJs());
			sb.append("  </script>\n");
		}
		sb.append("</body>\n");
		sb.append("</html>\n");
		return sb.toString();
	}
}